### PR TITLE
Force to use tls if startTls

### PR DESCRIPTION
**Pull request by stephdl**
The remote authentication to SAMBA AD is broken we must force the TLS and the use of port 636 if we detect that starttls is used with AD

```
@eldap:report_bind_failure:988 LDAP bind failed on nsdc-ns7loc12.ad.nethservertest.org:389
Reason: strongAuthRequired
```

Ejabberd don't use the protocol startTls, the TLS encryption is 'none|tls', only ldap over ssl is supported, if startTls is detected then we must force the tls encryption on a port you can customize.

https://github.com/NethServer/dev/issues/6399

### CODE REVIEW

**Review by gsanchietti — CHANGES_REQUESTED** [review]
I'm pretty sure that such change will break existing installations connected to Microsoft AD without TLS.

**Comment by stephdl** [comment]
Actually when we create a bind to a remote AD with cockpit we enable starttls by default hence our template understand we need to use starttls, however  when we install ejabberd directly on a NS and a local AD  account provider, we use tls and 636 port

**Review by DavidePrincipi — COMMENTED** [review]
What about adding a prop to gradually upgrade existing systems to this new behavior?

We did a similar change for Nextcloud here https://github.com/NethServer/dev/issues/6318

File: nethserver-ejabberd.spec [hunk 1]
```diff
@@ -69,7 +69,7 @@ mkdir -p %{buildroot}/%{_localstatedir}/lib/nethserver/ejabberd/upload
 %systemd_preun ejabberd.service
 
 %postun
-%systemd_postun
+%systemd_postun ejabberd.service
```
Comment (DavidePrincipi) [inline nethserver-ejabberd.spec:72]: Could you post the error message related to this change?

I'd like to split this change in a separate PR :thinking:

Comment (stephdl) [reply nethserver-ejabberd.spec:72]: this is the error

```
╰─➤  make-rpms *.spec
make-rpms [INFO] git describe: 1.8.2 2 g62fa10f
error: The %systemd_postun macro requires some arguments
Traceback (most recent call last):
  File "/usr/bin/spectool", line 441, in <module>
    exit(main())
  File "/usr/bin/spectool", line 366, in main
    spec = Spec(path)
  File "/usr/bin/spectool", line 207, in __init__
    self.spec = rpm.spec(self.path)
ValueError: can't parse specfile

error: The %systemd_postun macro requires some arguments
Traceback (most recent call last):
  File "/usr/bin/spectool", line 441, in <module>
    exit(main())
  File "/usr/bin/spectool", line 366, in main
    spec = Spec(path)
  File "/usr/bin/spectool", line 207, in __init__
    self.spec = rpm.spec(self.path)
ValueError: can't parse specfile

basename: missing operand
Try 'basename --help' for more information.
Cleaning up temporary files..
removed './.tmpO2liP99.spec'
removed './.tmpWQleDJI.gitlog'
```

It seems something new, but I am sure it makes sense

Comment (gsanchietti) [reply nethserver-ejabberd.spec:72]: Proposed change seems good to me, see https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#_systemd

I'm ok for moving to another PR

Comment (stephdl) [reply nethserver-ejabberd.spec]: this has been moved to https://github.com/NethServer/nethserver-ejabberd/pull/38

**Comment by stephdl** [comment]
Ejabberd don't use the protocol startTls, the TLS encryption is 'none|tls', only ldap over ssl is supported, if startTls is detected then we must force the tls encryption on a port you can customize.

File: README.rst [hunk 2]
```diff
@@ -35,7 +35,7 @@ Properties:
 * *ModMamPurgeDBInterval*: Remove messages older than X days, default is ``30``
 * *ShaperFast*: Download speed limit in bytes/second for admin users, default is ``1000000``
 * *ShaperNormal*: Download speed limit in bytes/second for users, default is ``500000``
-
+* *TlsPort*: The tls port of the remote ldap server can be customized (default is 636)
```
Comment (DavidePrincipi) [inline README.rst:38]:
```suggestion
* *LdapTlsPortOverride*: The user roster LDAP client does not support STARTTLS. If the system account provider is configured with STARTTLS, configure the user roster LDAP client to use this TCP port number during connections instead (default is port 636).
```

File: root/etc/e-smith/templates/etc/ejabberd/ejabberd.yml/85ModulesRoasterLdap [hunk 3]
```diff
@@ -15,6 +15,10 @@
         $ldap_encrypt = 'tls';
     }
 
+    if ($sssd->startTls()) {
+        $ldap_encrypt = 'tls';
+        $ldap_port = $ejabberd{'TlsPort'} || '636';
```
Comment (DavidePrincipi) [inline root/etc/e-smith/templates/etc/ejabberd/ejabberd.yml/85ModulesRoasterLdap:20]:
```suggestion
        $ldap_port = $ejabberd{'LdapTlsPortOverride'} || '636';
```

**Review by DavidePrincipi — COMMENTED** [review]
TlsPort is a bit generic. I'd like a longer one `LdapTlsPortOverride`.

**Review by gsanchietti — APPROVED** [review]
It looks good to me.
Let's wait @DavidePrincipi approval

**Review by DavidePrincipi — APPROVED** [review]
